sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -734,6 +766,13 @@ class StreamExecution(
   case object TERMINATED extends State
 }
 
+object StreamExecutionMetadata {
+  private implicit val formats = Serialization.formats(NoTypeHints)
+
+  def apply(json: String): StreamExecutionMetadata =
```

Comment:
Could you add some unit tests for this? E.g.,
```
    assert(StreamExecutionMetadata(0, 0) ===
      StreamExecutionMetadata("""{}"""))
    assert(StreamExecutionMetadata(1, 0) ===
      StreamExecutionMetadata("""{"currentEventTimeWatermark":1}"""))
    assert(StreamExecutionMetadata(0, 2) ===
      StreamExecutionMetadata("""{"currentBatchTimestamp":2}"""))
    assert(StreamExecutionMetadata(1, 2) ===
      StreamExecutionMetadata("""{"currentEventTimeWatermark":1,"currentBatchTimestamp":2}"""))
```
Then it's easy for us to find any incompatible changes in future.